src/api/actions.rs

```diff
@@ -1,8 +1,53 @@
 //! GitHub Actions
 use snafu::ResultExt;
 
+use crate::models::{ArtifactId, RepositoryId, RunId, workflows::WorkflowDispatch};
 use crate::{params, Octocrab};
-use crate::models::{ArtifactId, RepositoryId, RunId};
+
+pub struct WorkflowDispatchBuilder<'octo> {
+    crab: &'octo Octocrab,
+    owner: String,
+    repo: String,
+    workflow_id: String,
+    data: WorkflowDispatch,
+}
+
+impl<'octo> WorkflowDispatchBuilder<'octo> {
+    pub(crate) fn new(crab: &'octo Octocrab, owner: String, repo: String, workflow_id: String) -> Self {
+        Self { crab, owner, repo, workflow_id, data: Default::default() }
+    }
+
+    /// Required
+    pub fn ref_field(mut self, ref_attr: impl Into<String>) -> Self {
+        self.data.ref_field = ref_attr.into();
+        self
+    }
+
+    /// Optional
+    pub fn inputs(mut self, inputs: serde_json::Value) -> Self {
```

Comment:
This should document that it panics, when it does, and include information on what the property does. You can mostly just copy from GitHub's own REST documentation. https://docs.github.com/en/rest/reference/actions#create-a-workflow-dispatch-event

```suggestion
    /// Input keys and values configured in the workflow file. The maximum number of properties is 10.
    /// Any default properties configured in the workflow file will be used when inputs are omitted.
    ///
    /// # Panics
    /// If `inputs` is not `Value::Object`.
    pub fn inputs(mut self, inputs: serde_json::Value) -> Self {
```